File: atlantis-grev/backend/AtlantisGrev.API/Models/WhatsAppAccount.cs
using System.Text.Json.Serialization;

namespace AtlantisGrev.API.Models;

public class WhatsAppAccount
{
    [JsonPropertyName("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    
    [JsonPropertyName("user_id")]
    public long UserId { get; set; }
    
    [JsonPropertyName("phone_number")]
    public string PhoneNumber { get; set; } = string.Empty;
    
    [JsonPropertyName("status")]
    public AccountStatus Status { get; set; } = AccountStatus.Idle;
    
    [JsonPropertyName("warming_status")]
    public WarmingStatus WarmingStatus { get; set; } = WarmingStatus.NotStarted;
    
    [JsonPropertyName("session_dir")]
    public string SessionDir { get; set; } = string.Empty;
    
    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("warming_started_at")]
    public DateTime? WarmingStartedAt { get; set; }
    
    [JsonPropertyName("warming_completed_at")]
    public DateTime? WarmingCompletedAt { get; set; }
    
    [JsonPropertyName("warming_progress")]
    public int WarmingProgress { get; set; } = 0;
    
    [JsonPropertyName("warming_logs")]
    public List<string> WarmingLogs { get; set; } = new();
}

public enum AccountStatus
{
    Idle,
    Active,
    Warming,
    Completed,
    Failed,
    Suspended
}

public enum WarmingStatus
{
    NotStarted,
    Queued,
    InProgress,
    Paused,
    Completed,
    Failed
}

File: atlantis-grev/mobile/atlantis_grev_mobile/lib/models/whatsapp_account.dart
class WhatsAppAccount {
  final String id;
  final String phoneNumber;
  final String status;
  final String warmingStatus;
  final int warmingProgress;
  final DateTime createdAt;
  final DateTime? warmingStartedAt;
  final DateTime? warmingCompletedAt;
  final List<String>? warmingLogs;

  WhatsAppAccount({
    required this.id,
    required this.phoneNumber,
    required this.status,
    required this.warmingStatus,
    required this.warmingProgress,
    required this.createdAt,
    this.warmingStartedAt,
    this.warmingCompletedAt,
    this.warmingLogs,
  });

  factory WhatsAppAccount.fromJson(Map<String, dynamic> json) {
    return WhatsAppAccount(
      id: json['id'] as String? ?? '',
      phoneNumber: json['phoneNumber'] as String? ?? '',
      status: json['status'] as String? ?? 'Idle',
      warmingStatus: json['warmingStatus'] as String? ?? 'NotStarted',
      warmingProgress: (json['warmingProgress'] as num?)?.toInt() ?? 0,
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      warmingStartedAt: json['warmingStartedAt'] != null 
          ? DateTime.parse(json['warmingStartedAt']) 
          : null,
      warmingCompletedAt: json['warmingCompletedAt'] != null 
          ? DateTime.parse(json['warmingCompletedAt']) 
          : null,
      warmingLogs: json['warmingLogs'] != null 
          ? List<String>.from(json['warmingLogs']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'phoneNumber': phoneNumber,
      'status': status,
      'warmingStatus': warmingStatus,
      'warmingProgress': warmingProgress,
      'createdAt': createdAt.toIso8601String(),
      'warmingStartedAt': warmingStartedAt?.toIso8601String(),
      'warmingCompletedAt': warmingCompletedAt?.toIso8601String(),
      'warmingLogs': warmingLogs,
    };
  }

  // Helper methods
  bool get isWarming => warmingStatus == 'InProgress';
  bool get isCompleted => status == 'Completed';
  bool get isFailed => status == 'Failed';
  bool get canStartWarming => status == 'Idle' && warmingStatus == 'NotStarted';
  bool get canPauseWarming => warmingStatus == 'InProgress';
  bool get canResumeWarming => warmingStatus == 'Paused';
}

